modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryServiceImpl.java

```diff
@@ -579,19 +587,77 @@ private void _validateSubmissionLimit(long objectDefinitionId, User user)
 			return;
 		}
 
-		int count = objectEntryPersistence.countByU_ODI(
-			user.getUserId(), objectDefinitionId);
-		long maximumNumberOfGuestUserObjectEntriesPerObjectDefinition =
-			_objectConfiguration.
-				maximumNumberOfGuestUserObjectEntriesPerObjectDefinition();
+		if (FeatureFlagManagerUtil.isEnabled("LPS-192957")) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinitionId);
+
+			try {
+				_objectConfiguration =
+					_configurationProvider.getCompanyConfiguration(
+						ObjectConfiguration.class,
+						objectDefinition.getCompanyId());
+
+				if (_objectConfiguration == null) {
+					_objectConfiguration =
+						_configurationProvider.getSystemConfiguration(
+							ObjectConfiguration.class);
+				}
+			}
+			catch (ConfigurationException configurationException) {
+				throw new RuntimeException(configurationException);
+			}
+
+			LocalDate beginningDate = _getStartDateInterval();
+
+			long count =
+				objectEntryLocalService.getObjectEntriesCount(user.getUserId(),
+					objectDefinition, ObjectEntryTable.INSTANCE.createDate.gte(
+						Date.from(beginningDate.atStartOfDay(
```

Comment:
Maybe you can move this manipulation of the date to your private method **_getStartDate()**, what do you think? This way you would just pass the **startDate** variable to **gte** method.